components/TARGET_PSA/TARGET_TFM/COMPONENT_NSPE/interface/src/tfm_ns_lock_rtx.c

```diff
@@ -53,6 +56,20 @@ uint32_t tfm_ns_lock_dispatch(veneer_fn fn,
         return TFM_ERROR_GENERIC;
     }
 
+    /* When the following conditions all assert, we can see it as we have 
+     * acquired NS lock and do dispatch:
+     * 1. Not in IRQ mode
+     * 2. IRQ masked
+     * 3. NS lock not owned by any thread */
+    bool can_dispatch_with_irq_masked = (!IsIrqMode()) && IsIrqMasked() && (!ns_lock.lock_count);
+    if (can_dispatch_with_irq_masked) {
+        return fn(arg0, arg1, arg2, arg3);
+    }
+
+    if (core_util_atomic_incr_u16(&ns_lock.lock_count, 1) == 0) {
+        MBED_ERROR1(MBED_MAKE_ERROR(MBED_MODULE_UNKNOWN, MBED_ERROR_CODE_OVERFLOW), "NS lock count overflow (> 0xFFFF)", ns_lock.lock_count);
```

Comment:
Given that the increment and decrement are in the same function, having all these checks seems overkill. Other lock/unlock in Mbed OS has the checks because they're separate functions and could be mismatched by the caller.